src/main/java/com/purbon/kafka/topology/model/Topic.java

```diff
@@ -1,24 +1,37 @@
 package com.purbon.kafka.topology.model;
 
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.purbon.kafka.topology.TopologyBuilderConfig;
-import com.purbon.kafka.topology.model.Impl.TopicImpl;
 import com.purbon.kafka.topology.model.users.Consumer;
 import com.purbon.kafka.topology.model.users.Producer;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
-@JsonDeserialize(as = TopicImpl.class)
```

Comment:
why? as far as I know, this is need in order to tell jackson there is an implementation class. The thing is the API, uses another implementation, while this might not be needed in the future, I would like to keep it unless we can find a way to do the same. Hope it makes sense?

Reply:
Was aligning `Topic` with `Topology` interface. But found that `Project` also uses this approach.

Was confusing why `TopologyDeserializer<Topology>` and `TopicDeserializer<TopicImpl>`. This PR aligns these.

If this approach is preferred, I can turn it back, including `Topology`. Trying to help with consistency here.

Reply:
https://github.com/kafka-ops/kafka-gitops-api is the project I started to play with this idea and where this is used.